compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUPromoteMatmulToFitMMA.cpp

```diff
@@ -0,0 +1,167 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/LLVMGPU/PassDetail.h"
+#include "iree/compiler/Codegen/LLVMGPU/Passes.h"
+#include "iree/compiler/Dialect/Flow/IR/FlowOps.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/Dialect/MemRef/Transforms/Transforms.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#define DEBUG_TYPE "iree-llvmgpu-promote-matmul-to-fit-mma"
+
+namespace mlir::iree_compiler {
+#define GEN_PASS_DECL_LLVMGPUPROMOTEMATMULTOFITMMA
+#include "iree/compiler/Codegen/LLVMGPU/Passes.h.inc"
+namespace {
+
+class LLVMGPUPromoteMatmulToFitMMAPass
+    : public LLVMGPUPromoteMatmulToFitMMABase<
+          LLVMGPUPromoteMatmulToFitMMAPass> {
+public:
+  explicit LLVMGPUPromoteMatmulToFitMMAPass(
+      const LLVMGPUMatmulPadOption &option) {
+    this->targetDimensions.setValue(option);
+  }
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry.insert<tensor::TensorDialect, linalg::LinalgDialect>();
+  }
+
+  void pad(RewriterBase &rewriter, linalg::LinalgOp op,
```

Comment:
nit:
```suggestion
  void padContractionOp(RewriterBase &rewriter, linalg::LinalgOp op,
```
because the zero padding implicates a contraction op.

Reply:
I mis-track some trick about padding because we deprecated it for a long time on CPU path. IIRC, we could need to pad `linalg.fill` op as well. This is not happening now. How about renaming it to `padWithZeros`, which is more general?

Reply:
Sounds good to me